src/main/ml-modules/root/lib/facetsLib.mjs

```diff
@@ -16,75 +16,96 @@ import { BadRequestError } from './mlErrorsLib.mjs';
 import { FACETS_VIA_SEARCH_CONFIG } from '../config/facetsViaSearchConfig.mjs';
 import { facetToScopeAndTermName } from '../utils/searchTermUtils.mjs';
 
-// A safeguard from a current or future viaSearch facet request from pulling too many documents from disk.
-// Not a concern for the initial two facets but what about the next one?  As opposed to facets that pull
-// their values from a field, enabling a request to pull an unlimited number of documents from disk would
-// be bad.
-const MAXIMUM_FACET_VALUE_VIA_SEARCH_DOCUMENTS = 100;
+// Pagination constants
+const DEFAULT_PAGE = 1;
+const DEFAULT_PAGE_LENGTH = 20;
+const MAXIMUM_NON_SEMANTIC_PAGE_LENGTH = 10000;

```

Comment:
`MAXIMUM_SEMANTIC_PAGE_LENGTH` is presently used in two contexts:

1. Maximum page length, as the variable name convey.
2. Maximum number of results when searching for the facet values.

As such, one can never retrieve the 101st value for a semantic facet.  This was the case before pagination.  I'll update the API documentation to make this clear.

Reply:
Done